Library/Homebrew/cmd/audit.rb

```diff
@@ -161,6 +164,42 @@ def audit_class
 
   @@aliases ||= Formula.aliases
 
+  def audit_formula_name
+    return unless @strict
+    # skip for non-official taps
+    return if !formula.core_formula? && !formula.tap.to_s.start_with?("homebrew")
+
+    name = formula.name
+    full_name = formula.full_name
+
+    if @@aliases.include? name
+      problem "Formula name is conflicted with existed aliases"
+      return
+    end
+
+    homebrew_tapd = HOMEBREW_LIBRARY/"Taps/homebrew"
+    current_taps = if homebrew_tapd.directory?
+      homebrew_tapd.subdirs.map(&:basename).map { |tap| tap.to_s.sub(/^homebrew-/, "") }
+    else
+      []
+    end
+    missing_taps = OFFICIAL_TAPS - current_taps
+
+    begin
+      nostdoutstderr do
+        missing_taps.each { |tap| Homebrew.install_tap "homebrew", tap }
+      end
+      same_name_formulae = Formula.full_names.group_by { |n| n.split("/").last }.fetch(name)
+      if same_name_formulae.size > 1
+        problem "Formula name is conflicted with #{(same_name_formulae - [full_name]).join ", "}"
+      end
+    ensure
+      nostdoutstderr do
+        missing_taps.each { |tap| Homebrew.uninstall_tap "homebrew", tap }
+      end
```

Comment:
We definitely don't need to install each of these taps when we can use `brew search` instead.